test/minishift/test_suite.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/fabric8-services/fabric8-tenant/cluster"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/fabric8-services/fabric8-tenant/openshift"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/resource"
+	"github.com/fabric8-services/fabric8-tenant/test/stub"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestSuite is a base for tests using Minishift and gorm db
+type TestSuite struct {
+	gormsupport.DBTestSuite
+	ClusterService      *stub.ClusterService
+	AuthService         *stub.AuthService
+	Config              *configuration.Data
+	toReset             func()
+	minishiftUrl        string
+	minishiftAdminName  string
+	minishiftAdminToken string
+	minishiftUserName   string
+	minishiftUserToken  string
+}
+
+func (s *TestSuite) SetupTest() {
+	resource.Require(s.T(), resource.Database, "F8_MINISHIFT")
+	s.minishiftUrl = os.Getenv("F8_MINISHIFT_URL")
+	s.minishiftAdminName = os.Getenv("F8_MINISHIFT_ADMIN_NAME")
+	s.minishiftAdminToken = os.Getenv("F8_MINISHIFT_ADMIN_TOKEN")
+	s.minishiftUserName = os.Getenv("F8_MINISHIFT_USER_NAME")
+	s.minishiftUserToken = os.Getenv("F8_MINISHIFT_USER_TOKEN")
+	s.DBTestSuite.SetupTest()
+	s.Config, s.toReset = prepareConfig(s.T())
+
+	log.InitializeLogger(s.Config.IsLogJSON(), s.Config.GetLogLevel())
+
+	s.ClusterService = &stub.ClusterService{
+		APIURL: s.minishiftUrl,
+		User:   s.minishiftAdminName,
+		Token:  s.minishiftAdminToken,
+	}
+	s.AuthService = &stub.AuthService{
+		OpenShiftUsername:  s.minishiftUserName,
+		OpenShiftUserToken: s.minishiftUserToken,
+	}
+}
+
+func (s *TestSuite) TearDownTest() {
+	s.DBTestSuite.TearDownTest()
+	s.toReset()
+}
+
+func (s *TestSuite) GetClusterService() cluster.Service {
+	return s.ClusterService
+}
+
+func (s *TestSuite) GetAuthService(tenantID uuid.UUID) auth.Service {
+	s.AuthService.TenantID = tenantID
+	return s.AuthService
+}
+
+func (s *TestSuite) GetConfig() *configuration.Data {
+	return s.Config
+}
+
+func prepareConfig(t *testing.T) (*configuration.Data, func()) {
+	resetVars := test.SetEnvironments(
+		test.Env("F8_AUTH_TOKEN_KEY", "foo"),
+		test.Env("F8_API_SERVER_USE_TLS", "false"),
+		test.Env("F8_LOG_LEVEL", "error"),
+		test.Env("F8_KEYCLOAK_URL", "http://keycloak.url.com"))
+	config, resetConf := test.LoadTestConfig(t)
+	reset := func() {
+		resetVars()
+		resetConf()
+	}
+	return config, reset
+}
+
+func VerifyObjectsPresence(t *testing.T, mappedObjects map[string]environment.Objects, options openshift.ApplyOptions, version string) {
+	for ns, objects := range mappedObjects {
+		var wg sync.WaitGroup
+		for _, obj := range objects {
+			wg.Add(1)
+			go func(toMarkAsDone *sync.WaitGroup, t *testing.T) {
+				iterations := 0
+				defer toMarkAsDone.Done()
```

Comment:
I believe you can use the `wg` variable directly from here. No need to pass it as an arg named `toMarkAsDone`

Reply:
I'm afraid that not - at least I faced an issue with it some time ago. In addition, I need to use a pointer to it otherwise it would use just a copy of the value.

Reply:
hm... let me try that :-)

Reply:
Yes, you are right - fixed here ebc661b